### ISSUE TITLE
[bug] naming-convention/ref-name: false positive with immediate deref

### ISSUE THREAD

**Issue by danielrentz** [Rel1cx/eslint-react#1375]
### Describe the bug

Rule `naming-convention/ref-name` should not trigger on this code:
```js
const obj = useRef({}).current;
```

### Platform and versions

```Text
@eslint-react 2.5.0
```

### Stack trace

```Text

```
